python/simulation.py

```diff
@@ -5171,3 +5175,45 @@ def merge_subgroup_data(data):
     comm.Alltoallv(smsg, rmsg)
 
     return output
+
+class BinaryPartition(object):
+    """
+    Binary tree class used for specifying a cell partition of arbitrary sized chunks for use as the
+    `chunk_layout` parameter of the `Simulation` class object.
+    """
+    def __init__(self, data=None, split_dir=None, split_pos=None, left=None, right=None, id=None):
+        """
+        The constructor accepts three separate groups of arguments: (1) `data`: a list of lists where each
+        list entry is either (a) a node defined as `[ (split_dir,split_pos), left, right ]` for which `split_dir`
+        and `split_pos` define the splitting direction (i.e., `mp.X`, `mp.Y`, `mp.Z`) and position (e.g., `3.5`,
+        `-4.2`, etc.) and `left` and `right` are the two branches (themselves `BinaryPartition` objects)
+        or (b) a leaf with integer value `id` in the range [0, `num_chunks`-1] for the chunk id, (2) a node
+        defined using `split_dir`, `split_pos`, `left`, and `right`, or (3) a leaf with `id`. This input format
+        enables specifying the binary tree using either a single list for the entire tree or defining the
+        nodes and leaves individually.
+        """
+        self.split_dir = None
+        self.split_pos = None
+        self.id = None
```

Comment:
Maybe rename this to `proc` to be consistent with the rest of the source code's name for a process ID.